sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncItemResponse.java

```diff
@@ -6,38 +6,37 @@
 import com.azure.cosmos.implementation.ResourceResponse;
 import org.apache.commons.lang3.StringUtils;
 
-public class CosmosAsyncItemResponse extends CosmosResponse<CosmosItemProperties> {
-    private final CosmosAsyncItem itemClient;
-
-    CosmosAsyncItemResponse(ResourceResponse<Document> response, PartitionKey partitionKey,
-                            CosmosAsyncContainer container) {
+public class CosmosAsyncItemResponse<T> extends CosmosResponse<CosmosItemProperties> {
+    private final Class<T> itemClassType;
+    
+    CosmosAsyncItemResponse(ResourceResponse<Document> response, Class<T> klass) {
         super(response);
+        this.itemClassType = klass;
         String bodyAsString = response.getBodyAsString();
-        if (StringUtils.isEmpty(bodyAsString)) {
+        if (StringUtils.isEmpty(bodyAsString)){
             super.setProperties(null);
-            itemClient = null;
         } else {
             CosmosItemProperties props = new CosmosItemProperties(bodyAsString);
             super.setProperties(props);
-            itemClient = new CosmosAsyncItem(props.getId(), partitionKey, container);
         }
     }
 
     /**
-     * Gets the itemSettings
+     * Gets the resource .
      *
-     * @return the itemSettings
+     * @return the resource
      */
-    public CosmosItemProperties getProperties() {
-        return super.getProperties();
+    public T getResource(){
+        return super.getProperties().toObject(itemClassType);
```

Comment:
this will result in double serialization (which was fixed prior to this PR)

Reply:
Changed this to avoid double serialization. Now we directly use the resource string to convert to user object.